email-ballerina/Module.md

```diff
@@ -0,0 +1,324 @@
+    pollingInterval: 2,
+    port: 993,
+    secureSocket: {
+        // Transport level configuration
+    }
+};
+```
+
+By default, TLS/SSL is enabled as the default transport level security protocol and the certificate verification is set as required.
+This optional protocol definition can be configured with the `enum` field, `security`, in each of the configuration type described above.
+
+Options available with the field, `security` are as follows.
+
+1. `SSL` - As same as the default TLS/SSL protocol
+2. `START_TLS_NEVER` - Disables both TLS/SSL and STARTTLS protocols and allows only the unencrypted transport level communication
+3. `START_TLS_ALWAYS` - Makes it mandatory to use secure STARTTLS protocol
+4. `START_TLS_AUTO` - Enables STARTTLS protocol which would switch to unsecure communication mode if the secure STARTTLS mode is not available in the server
+
+Following is an example of using the field, `security` in the SMTP client with `START_TLS_AUTO` mode.
+
+```ballerina
+email:SmtpConfiguration smtpConfig = {
+    port: 587,
+    secureSocket: {
+        // Transport level configuration
+    },
+    security: START_TLS_AUTO
+};
+```
+
+Similarly, other client/listener configuration types can also be defined with the field `security`.
+
+**Note**: Make sure the port number is changed accordingly depending on the protocol used.
+
+Standard port numbers used for each of the protocol for each type of transport security are as given below.
+
+| Protocol/Security | SSL | STARTTLS | Unsecure |
+|-------------------|-----|----------|----------|
+| **SMTP**          | 465 | 587      | 25, 587  |
+| **POP3**          | 995 | 995      | 110      |
+| **IMAP4**         | 993 | 143, 993 | 143      |
+
+All the authentication is based on username/password credentials.
+
+Note that when the `'from` field is not provided in an `email:Message`, the `username` field of the initialization argument to the `email:SmtpClient` is set as the `from` address of an email to be sent with SMTP.
+
+## Message Content and Attachments
+
+An `email:Message` prepared to be sent, can have the text body content, `body`, and/or HTML body content, `htmlBody`.
+When emails are received with POP3 or IMAP, text email bodies and HTML bodies of the email are captured by `body` and `htmlBody` fields of `email:Message` respectively.
+
+When sending emails with SMTP, there are four options to specify email `attachments` in `email:Message`.
+
+1. With `email:Attachment` type which points to an attachment file along with its content-type
+2. With an array of `email:Attachment` type
+3. With `mime:Entity` type
+4. With an array of `mime:Entity` type
+
+Option 1 and 2 are designed for ordinary users to attach files from the local machine along with its content-type.
+Option 3 and 4 are designed for advanced users who have programming knowledge to define complex MIME typed data attachments.
+
+Following is an example of attaching a PDF file to an email with option 1.
```

Comment:
```suggestion
The following is an example of attaching a PDF file to an email with option 1.
```

Reply:
Mistakenly skipped committing. Then I added the fix again below.